mlrun/api/crud/model_endpoints.py

```diff
@@ -1034,45 +1182,254 @@ def build_kv_cursor_filter_expression(
+            address=mlrun.mlconf.v3io_framesd,
+            container=self.container,
+        )
+
+        try:
+            data = frames_client.read(
+                backend="tsdb",
+                table=events_path,
+                columns=["endpoint_id", *metrics],
+                filter=f"endpoint_id=='{endpoint_id}'",
+                start=start,
+                end=end,
+            )
+
+            # Fill the metrics mapping dictionary with the metric name and values
+            data_dict = data.to_dict()
+            for metric in metrics:
+                metric_data = data_dict.get(metric)
+                if metric_data is None:
+                    continue
+
+                values = [
+                    (str(timestamp), value) for timestamp, value in metric_data.items()
+                ]
+                metrics_mapping[metric] = mlrun.api.schemas.Metric(
+                    name=metric, values=values
+                )
+        except v3io_frames.errors.ReadError:
+            logger.warn("Failed to read tsdb", endpoint=endpoint_id)
+        return metrics_mapping
+
+
+class _ModelEndpointSQLStore(_ModelEndpointStore):
+    @abstractmethod
+    def write_model_endpoint(self, endpoint, update=True):
+        raise NotImplementedError
+
+    @abstractmethod
+    def update_model_endpoint(self, endpoint_id, attributes):
+        raise NotImplementedError
+
+    @abstractmethod
+    def delete_model_endpoint(self, endpoint_id):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_model_endpoint(
+        self,
+        metrics: typing.List[str] = None,
+        start: str = "now-1h",
+        end: str = "now",
+        feature_analysis: bool = False,
+        endpoint_id: str = None,
+        raise_for_status=None,
+    ):
+        raise NotImplementedError
+
+    def list_model_endpoints(
+        self, model: str, function: str, labels: typing.List, top_level: bool
+    ):
+        raise NotImplementedError
+
+
+def get_endpoint_target(project, access_key):
+    """
+    Getting the DB target type based on mlrun.config.model_endpoint_monitoring.store_type.
+
+    :param project:             The name of the project.
+    :param access_key:          Access key with permission to write to DB table.
+
+    :return: ModelEndpointTarget object. Using this object, the user can apply different operations on the
+             model endpoint record such as write, update, get and delete.
+    """
+
+    # Get store type value from ModelEndpointStoreType enum class
+    model_endpoint_store_type = ModelEndpointStoreType(
+        mlrun.mlconf.model_endpoint_monitoring.store_type
+    )
+
+    # Convert into model endpoint store target object
+    return model_endpoint_store_type.to_endpoint_target(project, access_key)
+
+
+class ModelEndpointStoreType(enum.Enum):
+    """Enum class to handle the different store type values for saving a model endpoint record."""
+
+    kv = "kv"
+    sql = "sql"
+
```

Comment:
Is this class is only used by the API? If yes, move it to `api/schemas/model_endpoints.py` if not, let's discuss it.